activation/activation_test.go

```diff
@@ -1438,3 +1425,36 @@ func TestWaitingToBuildNipostChallengeWithJitter(t *testing.T) {
 		require.Less(t, deadline, time.Now())
 	})
 }
+
+// Test if GetPositioningAtx disregards ATXs with invalid POST in their chain.
+// It should pick an ATX with valid POST even though it's a lower height.
+func TestGetPositioningAtxPicksAtxWithValidChain(t *testing.T) {
+	tab := newTestBuilder(t)
+
+	// Invalid chain with high height
+	sigInvalid, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	ch := newChallenge(1, types.EmptyATXID, tab.goldenATXID, postGenesisEpoch, &tab.goldenATXID)
+	nipostData := newNIPostWithChallenge(t, types.HexToHash32("for invalid"), []byte("66"))
```

Comment:
That's the same as

```suggestion
	nipostData := newNIPostWithChallenge(t, shared.ZeroChallenge, []byte("66"))
```

`types.HexToHash32` decodes hex encoded bytes to a `[32]byte`, silently failing and returning all zeros when the provided string is not a valid hex encoded value (prepending it with zeros if its shorter than 32 bytes and cutting of bytes at the end if its longer than 32 bytes).

The same is also true further down for the other `nipostData` value, that's probably why you couldn't use the same poetRef without ending up with colliding ATXIDs 🙂

Reply:
Right, I probably intended to pick BytesToHash instead :facepalm:. Thanks

Reply:
The problem with using same `poetRef` is different. The challenge passed to `newNIPostWithChallenge` makes **zero** difference. The membership proof doesn't contain the proven leaf (here the challenge) and hence it is identical no matter what challenge is passed. Normally, this is a hash of the nipostchallenge of an ATX and it is required to be passed to the function that validates the membership merkle proof as the proven leaf.

Something must have gone wrong during refactors or code changes. The challenge could be completely removed from `newNIPostWithChallenge` function without harm. I will do it in a separate PR later.

Reply:
Maybe the code should look like this? The most other tests seem to do this incorrectly as well.

```go
ch := newChallenge(1, types.EmptyATXID, tab.goldenATXID, postGenesisEpoch, &tab.goldenATXID)
nipostData := newNIPostWithChallenge(t, ch.Hash32(), []byte("66"))
```

Reply:
What shall we do with this now?

Reply:
Nothing. As we discussed I'm planning to refactor `newNIPostWithChallenge(` to not take "challenge" argument as it makes zero difference in the resulting `Nipost` in a separate PR.